modules/p2/ding_list/src/DingList.php

```diff
@@ -0,0 +1,620 @@
+      if ($element->getValue() == $value) {
+        return $element;
+      }
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get count of followers for a list.
+   *
+   * @return int
+   *   The followers count.
+   */
+  public function getFollowersCount() {
+    return count(ding_provider_invoke('openlist', 'call_module', 'Query', 'getLists', array(
+      $this->id,
+      array(DING_LIST_TYPE_LISTS),
+    )));
+  }
+
+  /**
+   * Get owner name of the list.
+   *
+   * @return string
+   *   The username.
+   */
+  public function getOwnerName() {
+    $owner = ding_provider_invoke('openlist', 'drupal_user', $this->owner);
+
+    $result = t('Another loaner');
+    if (!empty($owner->data) && isset($owner->data['display_name'])) {
+      $result = $owner->data['display_name'];
+    }
+    elseif (!empty($owner->realname)) {
+      $result = $owner->realname;
+    }
+
+    return check_plain($result);
+  }
+
+  /**
+   * Make sure you're allowed to perform a certain action on a list.
+   *
+   * @param string $operation
+   *   The operation you wish to perform.
+   *
+   * @return bool
+   *   TRUE if the list has the operation, and FALSE if it doesn't.
+   */
+  public function allowed($operation) {
+    $list_operations = ding_list_list_operations();
+    return isset($list_operations[$this->getType()][$operation]);
+  }
+
+  /**
+   * Move an element in a list down below another element.
+   *
+   * @param int $element_id
+   *   ID of the element ot move.
+   * @param int $previous
+   *   ID of the element to position the $element after.
+   *   If this is 0 the $element is positioned as the first element.
+   *
+   * @return bool
+   *   If the move is successful or not.
+   */
+  public function setElementPosition($element_id, $previous = 0) {
+    try {
+      ding_provider_invoke('openlist', 'set_element_after', $this->buildDataArray(), $element_id, $previous);
+    }
+    catch (Exception $e) {
+      watchdog_exception('ding_list', $e);
+      return FALSE;
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Add permissions to a list and user.
+   *
+   * @param string $permission
+   *   The permission to add.
+   * @param object $account
+   *   The user to give the permissions.
+   */
+  public function setPermission($permission, $account = NULL) {
+    if ($account === NULL) {
+      global $user;
+      $account = $user;
+    }
+
+    $openlist_uid = ding_provider_invoke('openlist', 'user_identifier', $account);
+    ding_provider_invoke('openlist', 'call_module', 'ListPermission', 'setPermission', array(
+      $openlist_uid,
+      $this->getId(),
+      $permission,
+    ), array('use_cache' => FALSE));
+
+    ding_provider_invoke('openlist', 'clear_cache', 'lp' . $this->getId());
+    $this->permissions[$account->uid] = $permission;
+  }
+
+  /**
+   * Remove a permission.
+   *
+   * @param object $account
+   *   The user to give the permissions.
+   */
+  public function removePermission($account = NULL) {
```

Comment:
The DocBlock is not up to date here.


I wonder why `setPermission` has an argument specifying which permission is added, but `removePermission` does not. Does`removePermission` remove all permissions for the user? Perhaps it is just my lack of understanding of the list structure.

Reply:
It's a singular permission, so a user is connected to 1 or more lists, but the permission is a simple, view/edit permission.
The `removePermission` simply removes that specific permission.